code-challenges.test.js

```diff
@@ -25,11 +53,61 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+const codeWord = (string) => {
+    return string.map(value => {
+         string.replaceAll("a", "4", "e", "3", "i", "1", "o", "0")
+    })
+}
+
+//--------------------------------------------------------------- QUESTION 1 NOTES---------------------------
+//--this is the error I got
+// ● codeWord › takes in a string and returns a coded message that converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+
+// TypeError: string.map is not a function
+
+//   42 |
+//   43 | const codeWord = (string) => {
+// > 44 |     let eachLetter = string.map(value => {
+//      |                             ^
+//   45 |         return string.replaceAll("a", "4", "e", "3", "i", "1", "o", "0")
+//   46 |     })
+//   47 | }
+
+//   at map (code-challenges.test.js:44:29)
+//   at Object.codeWord (code-challenges.test.js:24:16)
+//------------- Then I replaced that code by just getting rid of the 'let' and still got the same error 'string.map is not a function
+
```

Comment:
Your pseudocode is beautiful! You have a good idea -  The issue you are having is because you can't call .map on a string, it must be on an array.